docs/lsm/allocation/allocation_sources/allocation_v3/basic_example/v2_main.cf

```diff
@@ -0,0 +1,81 @@
+"""
+    Inmanta LSM
+    :copyright: 2024 Inmanta
+    :contact: code@inmanta.com
+    :license: Inmanta EULA
+"""
+
+import lsm
+import lsm::fsm
+
+entity TopLevelService extends lsm::ServiceEntity:
+    """
+    Top-level service to demonstrate V2 allocation.
+
+    :attr name: The name identifying the service instance.
+    :attr top_level_value: A read-only value, automatically assigned by the api.
+    """
+    string                      name
+    lsm::attribute_modifier     name__modifier="rw"
+    int?                        top_level_value=null
+    lsm::attribute_modifier     top_level_value__modifier="r"
+end
+
+# Enforce uniqueness based on the service's name
+index TopLevelService(name)
+
+# Each top level service may have zero or more embedded services attached to it
+TopLevelService.embedded_services [0:] -- EmbeddedService
+
+entity EmbeddedService extends lsm::EmbeddedEntity:
+    """
+    An embedded service, attached to a TopLevelService instance.
+
+    :attr id: The identifier of this embedded service instance, it should be unique
+        across all other embedded service instances of the service it is attached to.
```

Comment:
This is not aligned with the index behavior, the index says it should be unique across all instances of all services